docs/source/federation-spec.mdx

```diff
@@ -0,0 +1,625 @@
+
+For example, the `@apollo/subgraph` library defines the following resolver for `Query._entities`:
+
+```js
+{
+  _entities: (_source, { representations }, context, info) => entitiesResolver({ representations, context, info })
+}
+```
+
+This passes the list of representations to an `entitiesResolver` helper function, which is [defined here](https://github.com/apollographql/federation/blob/main/subgraph-js/src/types.ts#L64-L115).
+
+## Glossary of schema additions
+
+This section describes all of the types and field definitions that a valid subgraph service must automatically add to its schema. These definitions are all listed in [Subgraph schema specification](#subgraph-schema-specification).
+
+### == Added `Query` fields ==
+
+### `Query._service`
+
+This field of the root `Query` type must return a non-nullable [`_Service` type](#type-_service).
+
+For details, see [Enhanced introspection with `Query._service`](#enhanced-introspection-with-query_service).
+
+### `Query._entities`
+
+The graph router uses this root-level `Query` field to directly fetch fields of entities defined by a subgraph.
+
+This field must take a `representations` argument of type `[_Any!]!` (a non-nullable list of non-nullable [`_Any` scalars](#scalar-_any)). Its return type must be `[_Entity]!` (a non-nullable list of _nullable_ objects that belong to the [`_Entity` union](#union-_entity)).
+
+Each entry in the `representations` list  must be validated with the following rules:
+
+* A representation must include a `__typename` string field.
+* A representation must contain all fields included in the fieldset of a `@key` directive applied to the corresponding entity definition.
+
+For details, see [Resolving entity fields with `Query._entities`](#resolving-entity-fields-with-query_entities).
+
+### == Added types ==
+
+### `type _Service`
+
+This object type must have an `sdl: String!` field, which returns the SDL of the subgraph schema as a string.
+
+The returned schema string _must_ include all uses of federation-specific directives (`@key`, `@requires`, etc.), and it _must not_ include any definitions from the [subgraph schema specification](#subgraph-schema-specification).
+
+For details, see [Enhanced introspection with `Query._service`](#enhanced-introspection-with-query_service).
+
+### `union _Entity`
+
+**⚠️ This union type is generated dynamically based on the input subgraph schema!**
+
+This union's possible types must include all entities that the subgraph defines. It's the return type of the `Query._entities` field, which the graph router uses to directly access a subgraph's entity fields.
+
+For details, see [Defining the `_Entity` union](#defining-the-_entity-union).
+
+### `scalar _Any`
+
+This scalar is the type used for entity **representations** that the graph router passes to the `Query._entities` field. An `_Any` scalar is validated by matching its `__typename` and `@key` fields against entities defined in the subgraph schema.
```

Comment:
do we need to call out this is basically a map of field names/field value pairs (values could be scalars or objects)?